presto-parser/src/main/java/com/facebook/presto/sql/tree/DefaultTraversalVisitor.java

```diff
@@ -484,13 +482,30 @@ protected R visitGroupBy(GroupBy node, C context)
     }
 
     @Override
-    protected R visitGroupingElement(GroupingElement node, C context)
+    protected R visitCube(Cube node, C context)
```

Comment:
Is this correct behavior? Shouldn't we at least visit columns elements?

It's not that `DefaultTraversalVisitor#visitGroupingElement` was unused for `Cube` or `Rollup`. There are various classes that override: `AstVisitor#process(com.facebook.presto.sql.tree.Node, C)`. Are you sure we don't break any of them?

Reply:
I’ll double-check, but I think they are a no-op for those nodes. The only one that cares about it is SimpleGroupBy, which needs to get expressions analyzed/type-checked/etc.

The only thing I can think of that might break is column name resolution. I’ll verify that.

Reply:
Ok, it's not an issue. Column references are handled by looking at the enumerated expressions (related: https://github.com/prestodb/presto/issues/10478).